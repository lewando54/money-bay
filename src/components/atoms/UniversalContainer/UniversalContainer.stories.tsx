/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { Text, View } from 'react-native'
import { MaterialIcons } from '@expo/vector-icons'
import { Meta,StoryObj } from '@storybook/react-native'

import CircleWithIcon from '../CircleWithIcon/CircleWithIcon'

function ExampleChildren(){
    return (
        <View style={{justifyContent: 'space-between', alignItems: 'center', width: '100%', flexDirection: 'row'}}>
            <CircleWithIcon icon={<MaterialIcons name="compare-arrows" size={24} color="black" />}/>
            <Text>
                + 234.00 z≈Ç
            </Text>
        </View>
    )
}

import UniversalContainer from './UniversalContainer'

const meta = {
    component: UniversalContainer,
    argTypes: {
        width: { control: 'text' },
        padding: { control: 'number'},
        backgroundColor: { control: 'color' },
        castShadow: { control: 'boolean' }
    }
} satisfies Meta<typeof UniversalContainer>

export default meta
type Story = StoryObj<typeof UniversalContainer>

export const Default: Story = {
    args: {
        width: '100%'
    }
}

export const CastShadow: Story = {
    args: {
        width: '100%',
        castShadow: true
    }
}

export const CustomBackground: Story = {
    args: {
        width: '100%',
        backgroundColor: '#f00'
    }
}

export const WithChildren: Story = {
    args: {
        width: '100%',
        children: <ExampleChildren />
    }
}