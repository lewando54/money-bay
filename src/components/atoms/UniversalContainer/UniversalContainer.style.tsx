import { StyleSheet } from 'react-native'

interface IUniversalContainerStyleProps {
    width: string | number
    height: string | number
    padding: string | number
    backgroundColor: string
}

export default ({width, height, padding, backgroundColor}: IUniversalContainerStyleProps) => StyleSheet.create({
    container: {
        width: width,
        height: height,
        padding: padding,
        backgroundColor: backgroundColor,
        borderRadius: 10
    }
})