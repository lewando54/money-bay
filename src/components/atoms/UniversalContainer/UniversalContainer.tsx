import React from 'react'
import { StyleProp, TouchableOpacity, View, ViewStyle } from 'react-native'

import universalContainerStyle from './UniversalContainer.style'

import globalThemeStyle from '@styling/GlobalTheme.style'

interface IUniversalContainerProps {
    children?: React.ReactNode
    width?: string | number
    height?: string | number
    padding?: string | number
    backgroundColor?: string
    castShadow?: boolean
    style?: StyleProp<ViewStyle>
    onPress?: () => void
    testID?: string
}

export default function UniversalContainer({
    children,
    width='100%',
    height='auto',
    padding=10,
    backgroundColor='white',
    castShadow=false,
    style,
    onPress,
    testID='universal-container'
}: IUniversalContainerProps){

    if(onPress){
        return (
            <TouchableOpacity onPress={onPress}>
                <View
                    testID={testID} 
                    style={[
                        universalContainerStyle({width, height, padding, backgroundColor}).container,
                        castShadow && globalThemeStyle.default_Shadow,
                        style
                    ]}
                >
                    {children}
                </View>
            </TouchableOpacity>
        )
    }
    return (
        <View
            testID={testID} 
            style={[
                universalContainerStyle({width, height, padding, backgroundColor}).container,
                castShadow && globalThemeStyle.default_Shadow,
                style
            ]}
        >
            {children}
        </View>
    )
}