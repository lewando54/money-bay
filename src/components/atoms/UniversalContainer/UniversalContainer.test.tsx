import React from 'react'
import renderer from 'react-test-renderer'

import { CustomText } from '@atoms'

import UniversalContainer from './UniversalContainer'

describe('UniversalContainer', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<UniversalContainer />)
        expect(tree).toMatchSnapshot()
    })

    it('should render children', () => {
        const tree = renderer.create(<UniversalContainer><CustomText>Test</CustomText></UniversalContainer>)
        expect(tree).toMatchSnapshot()
    })

    it('should render with custom background color', () => {
        const tree = renderer.create(<UniversalContainer backgroundColor='#ff0' />)
        expect(tree).toMatchSnapshot()
    })

    it('should render with shadow', () => {
        const tree = renderer.create(<UniversalContainer castShadow={true} />)
        expect(tree).toMatchSnapshot()
    })
})