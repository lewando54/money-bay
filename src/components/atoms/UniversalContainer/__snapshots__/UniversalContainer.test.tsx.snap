// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversalContainer should render children 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
  testID="universal-container"
>
  <Text>
    Test
  </Text>
</View>
`;

exports[`UniversalContainer should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
  testID="universal-container"
/>
`;

exports[`UniversalContainer should render with custom background color 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ff0",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
  testID="universal-container"
/>
`;

exports[`UniversalContainer should render with shadow 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
  testID="universal-container"
/>
`;
